frontend/src/metabase/querying/components/TemporalUnitPicker/TemporalUnitPicker.tsx

```diff
@@ -0,0 +1,72 @@
+import { useState } from "react";
+import { t } from "ttag";
+
+import { Box, SelectItem } from "metabase/ui";
+import type { TemporalUnit } from "metabase-types/api";
+
+const MIN_WIDTH = 180;
+const INITIAL_VISIBLE_ITEMS_COUNT = 7;
+
+interface TemporalUnitItem {
+  value: TemporalUnit;
+  label: string;
+}
+
+interface TemporalUnitPickerProps {
+  value: TemporalUnit | undefined;
+  availableItems: TemporalUnitItem[];
+  canRemove?: boolean;
+  onChange: (newValue: TemporalUnit) => void;
+  onRemove?: () => void;
+}
+
+export function TemporalUnitPicker({
+  value,
```

Comment:
should we have a test when `value` is not defined? is it possible when no value is selected only?

Reply:
Added tests